libcontainer/cgroups/systemd/common.go

```diff
@@ -295,15 +296,23 @@ func generateDeviceProperties(rules []*devices.Rule) ([]systemdDbus.Property, er
 // and returns it
 func getDbusConnection(rootless bool) (*systemdDbus.Conn, error) {
 	connOnce.Do(func() {
-		if rootless {
-			connDbus, connErr = NewUserSystemdDbus()
-		} else {
-			connDbus, connErr = systemdDbus.New()
-		}
+		isRootless = rootless
+		connDbus, connErr = newDbusConnection(rootless)
 	})
 	return connDbus, connErr
 }
 
+func renewDbusConnection() {
+	connDbus, connErr = newDbusConnection(isRootless)
```

Comment:
This is a data race -- two threads could call `renewDbusConnection` simultaneously. If we need to set this conditionally (potentially multiple times) we probably need a mutex.

Reply:
(Another option would be https://pkg.go.dev/golang.org/x/sync/singleflight but that's probably overkill.)

Reply:
Updated, what do you think of this.